CONVENTION.md

```diff
@@ -0,0 +1,453 @@
+	- Pages are the full screen of an application, except statusbar, docks, toolbar and menu. Most apps only have one page, but some - like Pyblish Lite and QML - have more
+	- Panels are subdivisions of a Page
+	- Widgets are self explanatory
+- **S2** Widgets MUST be declared at the top
+- **S3** Widgets MUST have a mixedCase name, used in CSS
+- **S4** `Window` MUST have `title` declared as class attribute, used to dynamically change window title at run-time, e.g. to reflect a selection or state
+- **S5** Like widgets, icons SHOULD be declared up-front
+- **S6** Resource logic and I/O SHOULD be handled separately, for testing
+- **S7** Object names SHOULD be mixedCase, to conform with JavaScript/CSS where they are used
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+
+		...
+
+        for name, widget in itertools.chain(panels.items(),
+                                            widgets.items(),
+                                            pages.items()):
+            # Expose to CSS
+            widget.setObjectName(name)
+
+        	# Support for CSS
+            widget.setAttribute(QtCore.Qt.WA_StyledBackground)
+
+        self.setCentralWidget(widgets["pages"])
+        widgets["pages"].addWidget(pages["home"])
+
+```
+
+**Notes**
+
+- **S10** Every widget MUST be given a unique name
+- **S11** Every widget MUST be stylable with CSS
+- **S12** Central widget MUST be a `QStackedWidget` to facilitate multiple pages 
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+
+    	...
+
+    	layout = QtWidgets.QHBoxLayout(panels["header"])
+    	layout.setContentsMargins(0, 0, 0, 0)
+    	layout.setMargin(0)
+    	layout.addWidget(widgets["logo"])
+
+    	layout = QtWidgets.QHBoxLayout(panels["body"])
+    	layout.setContentsMargins(0, 0, 0, 0)
+    	layout.setMargin(0)
+    	layout.addWidget(widgets["okBtn"])
+    	layout.addWidget(widgets["cancelBtn"])
+    	layout.addWidget(widgets["resetBtn"])
+
+    	layout = QtWidgets.QHBoxLayout(panels["body"])
+    	layout.setContentsMargins(0, 0, 0, 0)
+    	layout.setMargin(0)
+    	layout.addWidget(QtWidgets.QLabel("My Footer"))
+
+    	#  ___________________
+    	# |           |       |
+    	# |___________|       |
+    	# |			  | 	  |
+    	# |___________|       |
+    	# |           |       |
+    	# |___________|_______|
+    	#
+    	layout = QtWidgets.QGridLayout(pages["home"])
+    	layout.setContentsMargins(0, 0, 0, 0)
+    	layout.addWidget(panels["header"], 0, 0)
+    	layout.addWidget(panels["body"], 1, 0)
+    	layout.addWidget(panels["footer"], 2, 0)
+    	layout.addWidget(panels["sidebar"], 0, 1, 0, 2)
+```
+
+**Notes**
+
+- **S20** Layouts MUST all be called `layout`; don't bother with unique names or maintaining reference to them
```

Comment:
How does this work if you need to do `layout.addLayout()`?

Reply:
You don't.

Every layout must have a widget.

Reply:
Then maybe one of the notes should also be to avoid the use of [`layout.addLayout`](https://doc.qt.io/qt-5/qboxlayout.html#addLayout) which is also discussed in the [tips on layouting for Qt](https://doc.qt.io/qt-5/layout.html#tips-for-using-layouts)